File: YuGiOhCard/Controller/FeedbackViewController.m
#import "FeedbackViewController.h"
#import "HttpUtils.h"
#import "UrlConsts.h"
#import "StringConsts.h"

@interface FeedbackViewController ()

@end

@implementation FeedbackViewController

- (id)initWithNibName:(NSString *)nibNameOrNil bundle:(NSBundle *)nibBundleOrNil
{
    self = [super initWithNibName:nibNameOrNil bundle:nibBundleOrNil];
    if (self) {
        // Custom initialization
    }
    return self;
}

- (void)viewDidLoad
{
    [super viewDidLoad];
    self.title = RIGHT_MENU_FEEDBACK;
    [self.txtFeedback becomeFirstResponder];
}

- (void)didReceiveMemoryWarning
{
    [super didReceiveMemoryWarning];
}

-(IBAction) sendClicked:(id)sender {
    [self.view endEditing:YES];
    NSString * _appver = [[NSBundle mainBundle] objectForInfoDictionaryKey:(NSString *)kCFBundleVersionKey];
    NSString * appver = [_appver stringByAddingPercentEscapesUsingEncoding:NSUTF8StringEncoding];
    NSString * _txt = self.txtFeedback.text;
    NSString * txt = [_txt stringByAddingPercentEscapesUsingEncoding:NSUTF8StringEncoding];
    UIDevice * dev = [UIDevice currentDevice];
    NSString * osver = [dev systemVersion];
    NSString * osname = [dev systemName];
    NSString * _osstr = [NSString stringWithFormat:@"%@ (%@)", osname, osver];
    NSString * osstr = [_osstr stringByAddingPercentEscapesUsingEncoding:NSUTF8StringEncoding];
    NSString * _uuid = [dev identifierForVendor].UUIDString;
    NSString * uuid = [_uuid stringByAddingPercentEscapesUsingEncoding:NSUTF8StringEncoding];

    NSString * param = [NSString stringWithFormat:PARAM_FEEDBACK, uuid, @"", txt, appver, osstr];
    HttpUtils * hu = [HttpUtils alloc];
    [hu get:[NSString stringWithFormat:@"%@?%@", URL_FEEDBACK, param]];
    
    [self.navigationController popViewControllerAnimated:YES];
}

@end

File: swift/YuGiOhCard/Controller/FeedbackViewController.swift
import UIKit

class FeedbackViewController: UIViewController {

    
    @IBOutlet var txtFeedback: UITextView?
    @IBOutlet var btnSend: UIBarButtonItem?
    
    var inited = false
    
    override func viewWillLayoutSubviews() {
        super.viewWillLayoutSubviews()
        if (!inited) {
            inited = true
            for temp in self.view.subviews {
                if (temp is UITextView) {
                    UIUtils.scaleComponent(temp)
                }
            }
        }
    }
    
    override func viewWillAppear(animated: Bool) {
        UIUtils.setStatusBar(true)
        UIUtils.setNavBar(self.navigationController!.navigationBar)
    }
    
    override func viewDidLoad() {
        super.viewDidLoad()
        inited = false
        self.title = RIGHT_MENU_FEEDBACK
        self.txtFeedback!.becomeFirstResponder()
    }

    override func didReceiveMemoryWarning() {
        super.didReceiveMemoryWarning()
    }
    

    @IBAction func sendClicked(sender: AnyObject) {
        self.view.endEditing(true)
        let _appver = ApplicationUtils.getAppVersion() as NSString
        let appver = _appver.stringByAddingPercentEscapesUsingEncoding(NSUTF8StringEncoding)
        let _txt = self.txtFeedback!.text as NSString
        let txt = _txt.stringByAddingPercentEscapesUsingEncoding(NSUTF8StringEncoding)
        let dev = UIDevice.currentDevice()
        let osver = dev.systemVersion
        let osname = dev.systemName
        let _osstr = "\(osname) (\(osver))" as NSString
        let osstr = _osstr.stringByAddingPercentEscapesUsingEncoding(NSUTF8StringEncoding)
        let _uuid = dev.identifierForVendor!.UUIDString;
        let uuid = (_uuid as NSString).stringByAddingPercentEscapesUsingEncoding(NSUTF8StringEncoding)
    
        let param = NSString(format: PARAM_FEEDBACK, uuid!, "", txt!, appver!, osstr!)
        let hu = HttpUtils()
        hu.get("\(URL_FEEDBACK)?\(param)")
        self.navigationController!.popViewControllerAnimated(true)
    }

}
